src/conn/conn_stat.c

```diff
@@ -96,24 +113,32 @@ __statlog_config(WT_SESSION_IMPL *session, const char **cfg, bool *runp)
 		FLD_SET(conn->stat_flags, WT_CONN_STAT_ON_CLOSE);
 
 	/*
-	 * Statistics logging configuration requires either a wait time or an
-	 * on-close setting.
-	 */
-	if (!*runp && !FLD_ISSET(conn->stat_flags, WT_CONN_STAT_ON_CLOSE))
-		return (0);
-
-	/*
-	 * If any statistics logging is done, this must not be a read-only
-	 * connection.
+	 * We don't allow the path to be reconfigured. The reason is MongoDB
+	 * allows admin privileges to reconfigure running WiredTiger instances,
+	 * but admin privileges may be different from the privileges used to
+	 * start MongoDB. There's no strong reason it's useful to reconfigure
```

Comment:
I'd prefer if this comment was generalized away from MongoDB. Maybe something like:

``
We don't allow the log path to be reconfigured due to security concerns. Any application that passes input strings directly to reconfigure would expose themselves to a security hole. The added utility of reconfiguring a statistics log path isn't worth the security risk.
``